core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -184,7 +184,7 @@ public Snapshot apply() {
   /**
    * Returns the snapshot summary from the implementation and updates totals.
    */
-  private Map<String, String> summary(TableMetadata previous) {
+  protected Map<String, String> summary(TableMetadata previous) {
```

Comment:
Why was this change needed?

Reply:
this was due to my unnecessarily elaborate `apply()` implementation.. this should get reverted now.